xla/stream_executor/cuda/cuda_asm_compiler.cc

```diff
@@ -305,9 +305,11 @@ static absl::StatusOr<std::string> FindPtxAsExecutable(
 
 absl::StatusOr<SemanticVersion> GetAsmCompilerVersion(
     std::string_view preferred_cuda_dir) {
-  TF_ASSIGN_OR_RETURN(std::string ptxas_path,
-                      FindPtxAsExecutable(preferred_cuda_dir));
-  return GetToolVersion(ptxas_path);
+  return ::xla::GetOnce([&]() -> absl::StatusOr<SemanticVersion> {
```

Comment:
In the internal review, I got a comment from @beckerhe:

This kind of caching doesn't take different values of preferred_cuda_dir into account. It might happen to work right now, but it's hack nonetheless.

I have cloned your PR and have reverted the changes in this file, and the util.h change it is using, as they are unrelated to the other changes anyway, and will get that uncontroversial part landed. Can you please create a separate PR for these other changes, and get it reviewed by Henning?

Reply:
Created https://github.com/openxla/xla/pull/16734